### PR TITLE
Use EDF 0.7

### PR DESCRIPTION
**Pull request by omus**
If need be we can support both EDF 0.6 and 0.7 at the same time

### CODE REVIEW

File: src/export_edf.jl
Comment (jrevels) [inline src/export_edf.jl:100]: What's the benefit to adding this parameter restriction? I could be missing something, but seems like this is an implementation detail and I don't see a correctness/performance benefit to surface/codify it?

for example, it feels like this code shouldn't break if the `EDF.Signal` constructor decided to e.g. promote provided encoded sample data to Int32 (not that that would happen, just kinda making a hypothetical point)

Comment (ararslan) [reply src/export_edf.jl:100]: This `Union` is the type of the `signals` field in an `EDF.File` object now that `EDF.Signal` is parameterized on the encoded sample type (for BDF support in EDF.jl v0.7). I think the purpose of adding it here is that you won't be able to construct an `EDF.File` programmatically with a `Union{EDF.AnnotationsSignal,EDF.Signal}`, since `convert` that's called automatically with inner constructors will fail.

**Review by kolia — COMMENTED** [review]
If I understand the underlying change to EDF, this should allow OndaEDF to read in BDF without any code changes, and the changes in this PR are so that when you export it'll always write EDF (?)

Should there be a test that we can actually read BDF into Onda?

**Comment by omus** [comment]
> Should there be a test that we can actually read BDF into Onda?

The scope of this change it just to have an OndaEDF.jl version which is compatible with the latest EDF.jl version.

**Review by kolia — APPROVED** [review]

**Comment by omus** [comment]
Rebased to address conflict and I also bumped the version to 0.9.2

**Review by jrevels — CHANGES_REQUESTED** [review]
Before merging, this requires https://github.com/beacon-biosignals/OndaEDF.jl/pull/35 to be merged into it to fix the problem uncovered in #34 

(also that needs a test)

**Review by jrevels — APPROVED** [review]

**Comment by jrevels** [comment]
Approved now that #35 is merged into this PR. will merge assuming tests pass

**Pull request merged**
